src/utils/aviris-planet-pair.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python3
+
+import argparse
+import json
+import pprint
+import copy
+
+from datetime import datetime, timedelta
+from dateutil import parser
+from shapely.geometry import MultiPolygon, shape
+from shapely.strtree import STRtree
+
+
+def cli_parser():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--aviris-ids', required=False, type=str, default=None)
+    parser.add_argument('--aviris-scenes', required=True, type=str)
+    parser.add_argument('--output', required=False, type=str, default=None)
+    parser.add_argument('--planet-scenes', required=True, type=str)
+    parser.add_argument('--timedelta', required=False, type=int, default=33)
+
+    parser.add_argument('--ignore-year', required=False, dest='ignore_year', action='store_true')
+    parser.set_defaults(ignore_year=False)
+    return parser
+
+
+def feature_to_shapely(feature):
+    s = shape(feature.get('geometry')).buffer(0)
+    s.ide = feature.get('id')
+    s.dt = parser.parse(feature.get('properties').get('datetime'))
+    return s
+
+
+def shapely_to_dict(s):
+    return {'id': s.ide, 'datetime': s.dt.isoformat()}
+
+
+def predicate(a, b, delta):
+    if a.intersects(b):
+        return (a.dt - b.dt <= delta) and (b.dt - a.dt <= delta)
+    else:
+        return False
+
+
+if __name__ == '__main__':
+
+    args = cli_parser().parse_args()
+
+    delta = timedelta(days=args.timedelta)
+
+    matches = []
+
+    # curl 'https://franklin.nasa-hsi.azavea.com/collections/planet-4band-cogs/items/?limit=1000'
+    with open(args.planet_scenes, 'r') as f:
+        planet = [feature_to_shapely(feature) for feature in json.load(f).get('features')]
+        if args.ignore_year:
+            for s in planet:
+                s.dt = s.dt.replace(year = 1776)
+    strtree = STRtree(planet)
+
+    # curl 'https://franklin.nasa-hsi.azavea.com/collections/aviris-l1-cogs/items/?limit=1000'
```

Comment:
Don't you want to try https://github.com/stac-utils/pystac-client? It will hopefully handle pagination for you in addition.